### PR TITLE
Added a push notification asking for feedback when the app crashes

### LINKED ISSUE #1663: Show a notification when the app crashes

### PR DESCRIPTION
**Pull request by jaigak**
Implements and closes #1663

### CODE REVIEW

**Comment by jaigak** [comment]
@yair100 Please don't merge this yet. I'm planning for adding buttons to the notification so that the user can easily access the log file and open an issue.

**Comment by Jaiganeshkumaran** [comment]
Here's an idea. Before showing the notification, we could try to save all the tabs to a file. When the user launches the app again, tabs would be recovered and everything will be ready to go. It's like 'Continue where you left off' but when an exception is thrown.

**Comment by yaichenbaum** [comment]
> I'm planning for adding buttons to the notification so that the user can easily access the log file and open an issue.

The ideal use of the notification would be to ask the user to open an issue on GitHub.

> Here's an idea. Before showing the notification, we could try to save all the tabs to a file. When the user launches the app again, tabs would be recovered and everything will be ready to go. It's like 'Continue where you left off' but when an exception is thrown.

This is a great idea.

**Comment by jaigak** [comment]
@yair100 Should we add a button to restart the app? It doesn't make sense because the notification is sent after the app crashes. Instead, we could open the when the user clicks on the notification itself. That's the default behavior.

**Comment by Jaiganeshkumaran** [comment]
Here's how it would look like:
திரைப்பிடிப்பு (11)

**Comment by generalguy41** [comment]
Bit of feedback; While I think this is a great idea and looks well implemented, should it maybe not be combined to one button or something else? As in, if the user clicks "Report this issue", it brings them to the github issue page, but then they have to go look for the log location, which is a couple extra steps. And on the other side, if they click "Open log location", then it opens the log location, they have the log, but have to figure out where to report the issue. Unless there's some way to not dismiss the notification without manually clicking the "x", but I'm not sure if that's a thing...

**Comment by Jaiganeshkumaran** [comment]
@generalguy41 The notification is always in the action center until the users clears it. A single button can't open GitHub and the log file.

**Comment by generalguy41** [comment]
@Jaiganeshkumaran Does clicking one of those buttons not dismiss the notification entirely? Or does it only commit the action and then dismiss it to the action center?

**Comment by Jaiganeshkumaran** [comment]
The screenshot is just a prototype. The latest commit does not include the buttons yet. Only the text.

**Comment by generalguy41** [comment]
Yea, I agree. That would probably be the best solution.

**Comment by Jaiganeshkumaran** [comment]
Since UWP background tasks don't support launching URLs or folders, these buttons will activate the app but also will do the action.

**Comment by Jaiganeshkumaran** [comment]
For some reason, the notification is fired at the starting because of two exceptions occur in DrivesManager. One exception is caught thus preventing from crash but the notification still appears. This also occurs in the debugger. I'm working on rewriting the block to prevent sending a notification. Until then no notification.

**Comment by Jaiganeshkumaran** [comment]
I fixed it. Getting 'System.UsedSpace' doesn't working for OneDrive so it threw an exception. Now I have restored the notification.

**Comment by Jaiganeshkumaran** [comment]
@generalguy41 Unfortunately, I learned that clicking the buttons will remove it from action center for ever. Instead we could remove those buttons and instead clicking on it will open a window that offers these options.

**Comment by generalguy41** [comment]
Perhaps another alternative would be just to have the button open the url to the github bug page, and reference where to get the log file from there in the placeholder text?

**Comment by Jaiganeshkumaran** [comment]
We could tell users how to access the log location when creating a new issue. Maybe Luke can modify the issue templates to teach users how to upload the log file because most users don't know and they don't upload the log file.

**Comment by yaichenbaum** [comment]
@Jaiganeshkumaran The notification should have an option to restore tabs and an option to report the issue. The report issue button should link to [here](https://github.com/files-community/files-uwp/issues/new/choose).

**Comment by Jaiganeshkumaran** [comment]
> @Jaiganeshkumaran The notification should have an option to restore tabs and an option to report the issue. The report issue button should link to [here](https://github.com/files-community/files-uwp/issues/new/choose).

I already added to button to open the issue chooser. Instead of having a button to restore why not make it the default behaviour. Clicking the report button itself starts a new instance. I've not worked on saving the tabs yet but I'll do that soon.

**Comment by Jaiganeshkumaran** [comment]
When clicking on the report button or simply opening a new instance will show a notification if it could restore the tabs.

File: Files.Package/Package.appxmanifest
Comment (tsvietOK) [inline Files.Package/Package.appxmanifest:55]: Don't change this file.

File: Files/Views/MainPage.xaml.cs
Comment (tsvietOK) [inline Files/Views/MainPage.xaml.cs:133]: Add translation.

File: Files/Views/MainPage.xaml.cs
Comment (tsvietOK) [inline Files/Views/MainPage.xaml.cs:193]: Add translation.

**Review by tsvietOK — CHANGES_REQUESTED** [review]
Resolve conflicts, please.

**Comment by Jaiganeshkumaran** [comment]
Although it seem to save open tabs, it prevents from the app from crashing. I will remove that and make it for continue where you left off.

File: Files/Views/MainPage.xaml.cs
Comment (Jaiganeshkumaran) [reply Files/Views/MainPage.xaml.cs:133]: This notification has been removed since tab's are no longer restored.

**Comment by Jaiganeshkumaran** [comment]
I have modified the issue templates in my branch to emphasize to upload the log file. It should be live after merging this pull request.

File: Files/Views/SettingsPages/About.xaml.cs
Comment (yaichenbaum) [inline Files/Views/SettingsPages/About.xaml.cs:24]: Why did you move the code to SettingsViewModel?

Comment (Jaiganeshkumaran) [reply Files/Views/SettingsPages/About.xaml.cs:24]: To keep it same across both the notification and about. In future if the link changes, we could change it easily from one place.

File: Files/App.xaml.cs
Comment (yaichenbaum) [inline Files/App.xaml.cs:376]: We shouldn't add code that is not being used, it would be better to add this later when it's ready to be used.

File: Files/Filesystem/DriveItem.cs
Comment (yaichenbaum) [inline Files/Filesystem/DriveItem.cs:68]: Are these changes necessary for this pull request?

File: Files/Views/MainPage.xaml.cs
Comment (yaichenbaum) [reply Files/Views/MainPage.xaml.cs:133]: It looks like the code is still in here, can you remove the code or add the translations?

**Review by yair100 — CHANGES_REQUESTED** [review]

File: Files/Filesystem/DriveItem.cs
Comment (Jaiganeshkumaran) [reply Files/Filesystem/DriveItem.cs:68]: Yes because actually two exceptions are thrown for certain drives which causes a error notification on startup.

File: Files/Views/MainPage.xaml.cs
Comment (yaichenbaum) [inline Files/Views/MainPage.xaml.cs:124]:
```suggestion
```

**Review by yair100 — DISMISSED** [review]

**Review by yair100 — DISMISSED** [review]

**Review by tsvietOK — CHANGES_REQUESTED** [review]

**Review by tsvietOK — APPROVED** [review]

**Pull request merged**
